src/argowrapper/routes/routes.py

```diff
@@ -355,10 +271,17 @@ def retry_workflow(
     request: Request,  # pylint: disable=unused-argument
 ) -> str:
     """retries a currently failed workflow"""
-
+    workflow_details = argo_engine.get_workflow_details(workflow_name, uid)
     try:
-        return argo_engine.retry_workflow(workflow_name, uid)
+        new_parameters = {}
+        for param in workflow_details.get("arguments").get("parameters"):
+            if param.get("name") and param.get("name") in VALID_PARAMETERS:
+                new_parameters[param.get("name")] = param.get("value")
```

Comment:
I think similar logic should be added to submit method as well. Or some other solution to avoid mismatches between the set of parameters that are accepted at submit vs the ones that are used in retry... 🤔